### ISSUE TITLE
Canceling In-App Purchase Throws Wrong Exception

### ISSUE THREAD

**Issue by ccocchiaro** [robotmedia/RMStore#143]
When a user is prompted to continue an In-App Purchase they initiate, canceling it throws the wrong exception ("Cannot connect to iTunes Store"), so the app has no way of knowing whether to report it to the user or not.  The following exception below is being thrown but a more appropriate one should be used when canceling a transaction.

RMStore: transaction failed with product TCM_KC_ADD3STUDENTS and error Error Domain=SKErrorDomain Code=2 "Cannot connect to iTunes Store" UserInfo=0x17407b740 {NSLocalizedDescription=Cannot connect to iTunes Store}

**Comment by msmollin** [comment]
This seems to actually be a bug in StoreKit. 

`- (void)didFailTransaction:(SKPaymentTransaction *)transaction queue:(SKPaymentQueue*)queue error:(NSError*)error` 

is being called correctly from the result of SKPaymentTransactionObserver in `- (void)paymentQueue:(SKPaymentQueue *)queue updatedTransactions:(NSArray *)transactions` 

which contains a value of `SKPaymentTransactionStateFailed` when you tap cancel. 

I can confirm this happens on iOS 8.2 on device as well. Haven't upgraded my device yet to 8.3, but judging from the patch notes they probably didn't fix it there either. If you have time, I'd open a Radar and link it back here.

**Comment by pahnev** [comment]
Same here, but for some reason I have to cancel twice. 
And as far as I know I'm calling the transaction only once:
```objective-c
[[RMStore defaultStore] addPayment:self.card.productID success:^(SKPaymentTransaction *transaction) {
    [UIApplication sharedApplication].networkActivityIndicatorVisible = NO;
} failure:^(SKPaymentTransaction *transaction, NSError *error) {
    [UIApplication sharedApplication].networkActivityIndicatorVisible = NO;
    UIAlertView *alerView = [[UIAlertView alloc] initWithTitle:NSLocalizedString(@"Payment Transaction Failed", @"")
                                                       message:error.localizedDescription
                                                      delegate:nil
                                             cancelButtonTitle:NSLocalizedString(@"OK", @"")
                                             otherButtonTitles:nil];
    [alerView show];
}];
```

**Comment by msmollin** [comment]
@pahnev good call. That appears to work, although wow the localized description is REALLY not aligned with that enum naming convention.
